api/proto/teleport/provisioning/v1/provisioning.proto

```diff
@@ -104,4 +104,12 @@ message PrincipalStateStatus {
 
   // Error holds a description of the last provisioing error, if any.
   string error = 4;
+
+  // Revision hold the revision of the principal record provisioned into the
+  // downstream system
+  string revision = 6;
```

Comment:
can't we use the `revision` field from the metadata field from the `PrincipalState` ?

Reply:
The revision in the `PrincipalState` metadata is the revision of the provisioning state record. This is the revision of the principal that was provisioned downstream. Perhaps it should be called `provisioned_revision` instead?

Reply:
Gotcha. `provisioned_principal_revision` would be better in that case.

(though I am still thinking why would we need to maintain this value. Maybe explain that in a comment?)

Reply:
Initially it's a debugging tool, so I can check that the provisioned principal is the latest revision.

I'm hoping that it will make it easier to detect changes in the principal that require re-provisioning on service startup, but that's out of scope for now.